[BUG] YukiHookDataChannel. Supporting Android 14

It looks like DataChannel does not support Android 14. Crashes when trying to register receivers on YukiHookDataChannel class line 211.
Possible fix - https://github.com/TheranicaRD-SW/Fetch/commit/f82b752862e716b1249cebebf36bc708b65c9503

targetSdk = 34
YukiHookAPI 1.1.11

Stacktrace:
FATAL EXCEPTION: main
Process: ru.bluecat.alfabankpatcher, PID: 6272
java.lang.RuntimeException: Unable to create application ru.bluecat.alfabankpatcher.ui.App: java.lang.SecurityException: ru.bluecat.alfabankpatcher: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7003)
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2236)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8177)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
Caused by: java.lang.SecurityException: ru.bluecat.alfabankpatcher: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
	at android.os.Parcel.createException(Parcel.java:3041)
	at android.os.Parcel.readException(Parcel.java:3024)
	at android.os.Parcel.readException(Parcel.java:2966)
	at android.app.IActivityManager$Stub$Proxy.registerReceiverWithFeature(IActivityManager.java:5684)
	at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1852)
	at android.app.ContextImpl.registerReceiver(ContextImpl.java:1792)
	at android.app.ContextImpl.registerReceiver(ContextImpl.java:1780)
	at android.content.ContextWrapper.registerReceiver(ContextWrapper.java:755)
	at com.highcapable.yukihookapi.hook.xposed.channel.YukiHookDataChannel.register$yukihookapi_release(YukiHookDataChannel.kt:211)
	at com.highcapable.yukihookapi.hook.xposed.channel.YukiHookDataChannel.register$yukihookapi_release$default(YukiHookDataChannel.kt:208)
	at com.highcapable.yukihookapi.hook.xposed.application.ModuleApplication.onCreate(ModuleApplication.kt:86)
	at ru.bluecat.alfabankpatcher.ui.App.onCreate(App.kt:16)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1316)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6998)
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2236) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:205) 
	at android.os.Looper.loop(Looper.java:294) 
	at android.app.ActivityThread.main(ActivityThread.java:8177) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.ActivityManagerService.registerReceiverWithFeature(ActivityManagerService.java:13908)
	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:2570)
	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2720)
	at android.os.Binder.execTransactInternal(Binder.java:1339)
	at android.os.Binder.execTransact(Binder.java:1275)

--- a/yukihookapi/src/main/java/com/highcapable/yukihookapi/hook/xposed/channel/YukiHookDataChannel.kt
+++ b/yukihookapi/src/main/java/com/highcapable/yukihookapi/hook/xposed/channel/YukiHookDataChannel.kt
@@ -208,11 +208,15 @@ class YukiHookDataChannel private constructor() {
     internal fun register(context: Context?, packageName: String = context?.packageName ?: "") {
         if (YukiHookAPI.Configs.isEnableDataChannel.not() || context == null) return
         receiverContext = context
-        context.registerReceiver(
-            handlerReceiver, IntentFilter().apply {
-                addAction(if (isXposedEnvironment) hostActionName(packageName) else moduleActionName(context))
-            }
-        )
+
+        val filter = IntentFilter().apply {
+            addAction(if (isXposedEnvironment) hostActionName(packageName) else moduleActionName(context))
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            context.registerReceiver(handlerReceiver, filter, Context.RECEIVER_EXPORTED)
+
+        } else context.registerReceiver(handlerReceiver, filter)
+
         /** 排除模块环境下模块注册自身广播 */
         if (isXposedEnvironment.not()) return
         nameSpace(context, packageName).with {

--- a/yukihookapi/src/main/java/com/highcapable/yukihookapi/hook/xposed/parasitic/AppParasitics.kt
+++ b/yukihookapi/src/main/java/com/highcapable/yukihookapi/hook/xposed/parasitic/AppParasitics.kt
@@ -299,12 +299,16 @@ internal object AppParasitics {
                                  * @param result 回调 - ([Context] 当前实例, [Intent] 当前对象)
                                  */
                                 fun IntentFilter.registerReceiver(result: (Context, Intent) -> Unit) {
-                                    it.registerReceiver(object : BroadcastReceiver() {
+                                    val receiver = object : BroadcastReceiver() {
                                         override fun onReceive(context: Context?, intent: Intent?) {
                                             if (context == null || intent == null) return
                                             result(context, intent)
                                         }
-                                    }, this)
+                                    }
+                                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                                        it.registerReceiver(receiver, this, Context.RECEIVER_EXPORTED)
+
+                                    } else it.registerReceiver(receiver, this)
                                 }
                                 hostApplication = it
                                 AppLifecycleCallback.onCreateCallback?.invoke(it)

